src/pages/en/guides/imports.md

```diff
@@ -39,15 +39,25 @@ JavaScript can be imported using normal ESM `import` & `export` syntax. This wor
 ## TypeScript
 
 ```js
-import { getUser } from './user.ts';
-import type { UserType } from './user.ts';
+import { getUser } from './user';
+import type { UserType } from './user';
 ```
 
-Astro includes built-in support for [TypeScript](https://www.typescriptlang.org/). You can import `.ts` and `.tsx` files directly in your Astro project, and even write TypeScript code directly inside your [Astro component](/en/core-concepts/astro-components/#the-component-script).
+Astro includes built-in support for [TypeScript](https://www.typescriptlang.org/). You can import `.ts` and `.tsx` files directly in your Astro project, and even write TypeScript code directly inside your [Astro component script](/en/core-concepts/astro-components/#the-component-script) and any [hoisted script tags](/en/core-concepts/astro-components/#client-side-scripts).
 
-**Astro doesn't perform any type checking itself.** Type checking should be taken care of outside of Astro, either by your IDE or through a separate script. The [Astro VSCode Extension](/en/editor-setup/) automatically provides TypeScript hints and errors in your open files.
+**Astro doesn't perform any type checking itself.** Type checking should be taken care of outside of Astro, either by your IDE or through a separate script. For type checking Astro files, the [`astro check` command](/en/reference/cli-reference/#astro-check) is provided.
 
-📚 Read more about [TypeScript support in Astro.](/en/guides/typescript/)
+:::note[TypeScript and file extensions]
+Per [TypeScript's module resolution rules](https://www.typescriptlang.org/docs/handbook/module-resolution.html), the `.ts` or `.tsx` file extensions should not be used when importing TypeScript files. Instead TypeScript recommend to either use the `.js` and `.jsx` file extensions respectively or completely omit the file extension
```

Comment:
Trying to decide whether “TypeScript” is plural or singular for conjugation purposes makes my head hurt, so I’ve rewritten that bit and tried to simplify a little as well.

```suggestion
Per [TypeScript's module resolution rules](https://www.typescriptlang.org/docs/handbook/module-resolution.html), `.ts` and `.tsx` file extensions should not be used when importing TypeScript files. Instead, either use `.js`/`.jsx` file extensions or completely omit the file extension.
```